pkg/apis/openebs.io/v1alpha1/types.go

```diff
@@ -142,3 +150,77 @@ type Task struct {
 	// Kind is the kind corresponding to the task's actual content
 	Kind string `json:"kind"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+// +resource:path=cstorpool
+
+// CStorPool describes a cstor pool resource created as custom resource.
+type CStorPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
+
+	Spec CStorPoolSpec `json:"spec"`
+}
+
+// CStorPoolSpec is the spec listing fields for a CStorPool resource.
+type CStorPoolSpec struct {
+	Disks    DiskAttr      `json:"disks"`
+	PoolSpec CStorPoolAttr `json:"poolSpec"`
+}
+
+// DiskAttr stores the disk related attributes.
+type DiskAttr struct {
+	DiskList []string `json:"diskList"`
+}
+
+// CStorPoolAttr is to describe zpool related attributes.
+type CStorPoolAttr struct {
+	PoolName  string `json:"poolName"`
+	Cachefile string `json:"cachefile"`
+	PoolType  string `json:"poolType"` //mirror, striped
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorpools
+
+// CStorPoolList is a list of CStorPoolList resources
+type CStorPoolList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorPool `json:"items"`
+}
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorvolumereplica
+// +genclient:nonNamespaced
+
+// CStorVolumeReplica describes a cstor pool resource created as custom resource
+type CStorVolumeReplica struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
```

Comment:
Double quotes should come at the end for whole json tag like `json:"metadata ,omitempty"`

Reply:
will change